src/scenechange/mod.rs

```diff
@@ -308,8 +308,12 @@ impl<T: Pixel> SceneChangeDetector<T> {
       let frame2_scaled = frame2.planes[0].downscale(self.scale_factor);
       self.frame_buffer.push(frame2_scaled);
     } else {
-      self.frame_buffer.remove(0);
-      self.frame_buffer.push(frame2.planes[0].downscale(self.scale_factor));
+      // swap the first and second elements (instead of removing the first and then
+      // pushing a new frame) so that we can reuse the allocation of the `Plane<T>`
+      // inside `self.frame_buffer`
+      self.frame_buffer.swap(0, 1);
+      frame2.planes[0]
```

Comment:
I don't think this ever actually puts the downscaled frame2 onto the buffer, unless I'm mistaken. The frames in the buffer are still the previous two frames, just swapped in order, and frame2 is downscaled then thrown away.

Reply:
The previous code removed the first element and then pushed the new frame into the buffer, which also causes the second element to become the first element, and then the new frame is the second element. Here, we are swapping the frames (which also achieves causes the second element to become the first element), but then are reusing the already-existing `Plane<T>` to store the downscaled frame, which is written to by `downscale_in_place` since it takes a mutable reference to the plane. I could also be mistaken, but I don't think the downscaled frame is thrown away.

Reply:
Ah, I see. downscale_in_place is taking the buffer as an argument. So I was reading the code wrong.